Shared/Samples/Filter by definition expression or display filter/FilterByDefinitionExpressionOrDisplayFilterView.swift

```diff
@@ -0,0 +1,184 @@
+                        // Updates the feature count when the map view finishes drawing.
+                        guard drawStatus == .completed else { return }
+                        
+                        // Creates an envelope from the frame of the map view.
+                        let viewRect = geometryProxy.frame(in: .local)
+                        guard let viewExtent = mapViewProxy.envelope(
+                            fromViewRect: viewRect
+                        ) else { return }
+                        
+                        do {
+                            // Gets the feature count contained within the envelope.
+                            featureCount = try await model.queryFeatureCount(extent: viewExtent)
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .errorAlert(presentingError: $error)
+            }
+        }
+        .overlay(alignment: .top) {
+            Text("Feature count: \(featureCount)")
+                .multilineTextAlignment(.center)
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(8)
+                .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                Picker("Filter Mode", selection: $selectedFilterMode) {
+                    ForEach(FilterMode.allCases, id: \.self) { filterMode in
+                        Text(filterMode.label)
+                    }
+                }
+                .pickerStyle(.segmented)
+                .onChange(of: selectedFilterMode) { newFilterMode in
+                    // Filters the feature layer based on the new filter mode.
+                    model.filterFeatureLayer(filterMode: newFilterMode)
+                }
+            }
+        }
+    }
+}
+
+private extension FilterByDefinitionExpressionOrDisplayFilterView {
+    /// The view model for the sample.
+    final class Model: ObservableObject {
+        /// A map with a topographic basemap.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            
+            // Initially centers the map on San Fransisco, CA, USA.
+            map.initialViewpoint = Viewpoint(
+                center: Point(latitude: 37.7762, longitude: -122.4522),
+                scale: 7e4
+            )
+            return map
+        }()
+        
+        /// A feature layer of San Fransisco 311 incidents.
+        private let featureLayer: FeatureLayer = {
+            // Creates the feature layer from a feature service URL.
+            let featureTable = ServiceFeatureTable(url: .sanFransiscoIncidentsFeatureLayer)
+            return FeatureLayer(featureTable: featureTable)
+        }()
+        
+        /// The parameters for querying the feature layer.
+        private let queryParameters = QueryParameters()
```

Comment:
There isn't really a need to re-use the query parameters. Personally, I would just create them where needed as needed.